apps/studio/components/interfaces/Auth/Hooks/CreateHookSheet.tsx

```diff
@@ -300,52 +328,107 @@ export const CreateHookSheet = ({
-                  )}
-                />
-                <FormField_Shadcn_
-                  key="postgresValues.functionName"
-                  control={form.control}
-                  name="postgresValues.functionName"
-                  render={({ field }) => (
-                    <FormItemLayout
-                      label="Function name"
-                      description="Postgres function which will be called by Supabase Auth each time the hook is triggered."
-                    >
-                      <FormControl_Shadcn_>
-                        <FunctionSelector
-                          size="small"
-                          schema={values.postgresValues.schema}
-                          value={field.value}
-                          onChange={field.onChange}
-                          disabled={field.disabled}
-                        />
-                      </FormControl_Shadcn_>
-                    </FormItemLayout>
-                  )}
-                />
-              </div>
+              <>
+                <div className="grid grid-cols-2 gap-8 px-8">
+                  <FormField_Shadcn_
+                    key="postgresValues.schema"
+                    control={form.control}
+                    name="postgresValues.schema"
+                    render={({ field }) => (
+                      <FormItemLayout
+                        label="Postgres Schema"
+                        description="Postgres schema where the function is defined."
+                      >
+                        <FormControl_Shadcn_>
+                          <SchemaSelector
+                            size="small"
+                            showError={false}
+                            selectedSchemaName={field.value}
+                            onSelectSchema={(name) => field.onChange(name)}
+                            disabled={field.disabled}
+                          />
+                        </FormControl_Shadcn_>
+                      </FormItemLayout>
+                    )}
+                  />
+                  <FormField_Shadcn_
+                    key="postgresValues.functionName"
+                    control={form.control}
+                    name="postgresValues.functionName"
+                    render={({ field }) => (
+                      <FormItemLayout
+                        label="Function name"
+                        description="Postgres function which will be called by Supabase Auth each time the hook is triggered."
+                      >
+                        <FormControl_Shadcn_>
+                          <FunctionSelector
+                            size="small"
+                            schema={values.postgresValues.schema}
+                            value={field.value}
+                            onChange={(value) => {
+                              setStatements([])
+                              if (hook.method.type === 'postgres') {
```

Comment:
No need for this magic. Move all of this into a `useMemo` and use the `values` from line 252 to construct the statements.

Reply:
@ivasilov might need some help with this one - i can't seem to figure out why the `useMemo` doesn't recalculate when the functions selector change multiple times

Reply:
Fixed.